src/python/pants/util/dirutil.py

```diff
@@ -326,6 +335,13 @@ def safe_concurrent_rename(src, dst):
   """
   # Delete dst, in case it existed (with old content) even before any concurrent processes
   # attempted this write. This ensures that at least one process writes the new content.
+
+  # For .pants.d directory, since it is a symlink, use the base dir('/tmp/.pants.d') instead
+  if src.endswith(get_repo_root() + '/.pants.d'):
+    src = '/tmp/.pants.d'
```

Comment:
Could this be configurable and/or have a better default? /tmp is typically a tmpfs filesystem that doesn't persist across (among other things) a reboot.

Reply:
Yes, the plan is to add something like `pants_physical_workdir: /path/to/physical_dir` in `pants.ini` once this  prototyping is complete.  `'/tmp/.pants.d'` is being used for the sake of simplicity.